nixpkgs_review/nix.py

```diff
@@ -44,15 +45,91 @@ def nix_shell(
     cache_directory: Path,
     system: str,
     run: Optional[str] = None,
+    sandbox: bool = False,
 ) -> None:
+    nix_shell = shutil.which("nix-shell")
+    if not nix_shell:
+        raise RuntimeError("nix-shell not found in PATH")
+
     shell = cache_directory.joinpath("shell.nix")
     write_shell_expression(shell, attrs, system)
-    args = ["nix-shell", str(shell)]
+    if sandbox:
+        args = _nix_shell_sandbox(nix_shell, shell)
+    else:
+        args = [nix_shell, str(shell)]
     if run:
         args.extend(["--run", run])
     sh(args, cwd=cache_directory, check=False)
 
 
+def _nix_shell_sandbox(nix_shell: str, shell: Path) -> List[str]:
+    bwrap = shutil.which("bwrap")
+    if not bwrap:
+        raise RuntimeError(
+            "bwrap not found in PATH. Install it to use '--sandbox' flag."
+        )
+
+    warn("Using sandbox mode. Some things may break!")
+
+    def bind(
+        path: Union[Path, str],
+        ro: bool = True,
+        dev: bool = False,
+        try_: bool = False,
+    ) -> List[str]:
+        if dev:
+            prefix = "--dev-"
+        elif ro:
+            prefix = "--ro-"
+        else:
+            prefix = "--"
+
+        suffix = "-try" if try_ else ""
+
+        return [prefix + "bind" + suffix, str(path), str(path)]
+
+    def tmpfs(path: Union[Path, str]) -> List[str]:
+        return ["--tmpfs", str(path)]
```

Comment:
Since these don't capture any local variables they can be moved out of the function.